### PR TITLE
cilium: insert new backend IPs into neigh table

### PR DESCRIPTION
**Pull request by borkmann**
See commit desc /cc @brb

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8608)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
Travis, hitting same issue as on master:
```FAIL: <autogenerated>:1: StoreConsulSuite.TestStoreOperations
store_test.go:204:
    c.Assert(expect(func() bool { return localKey2.deleted() == 0 }), IsNil)
... value *errors.errorString = &errors.errorString{s:"timeout while waiting for expected value"} ("timeout while waiting for expected value")
level=info msg="Waiting for consul to elect a leader" subsys=kvstore
```

File: pkg/datapath/linux/neigh/neigh.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+)
+
+func NeighAddAddress(address net.IP) error {
+	link, err := route.NodeDeviceWithDefaultRoute()
+	if err != nil {
+		logrus.WithError(err).Error("Unable to retrieve default route device")
+		return err
+	}
+	// XXX arping lib sucks in that we need such lookup twice
+	iface, err := net.InterfaceByName(link.Attrs().Name)
```
Comment (brb) [inline pkg/datapath/linux/neigh/neigh.go]: Shouldn't we do a lookup via `option.Config.Device` in the case we pass `--device=<..>` to cilium-agent?

**Review by brb — COMMENTED** [review]
LGTM, just one question.

As follow-up (discussed offline):

1. Implement our own arping to avoid doing the link lookup twice and to avoid not needed functionality.
2. Extend `lbmap.UpdateService` to accept a callback. In the case of NodePort we would pass `func (backendIP net.IP) { NeighAddAddress(backendIP) } as the callback. This would allow to avoid calling the function for backends which do not belong to a NodePort service.

File: pkg/datapath/linux/neigh/neigh.go (hunk 1 above)
Comment (borkmann) [reply pkg/datapath/linux/neigh/neigh.go]: Agree, true. I'll fix.

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Pull request merged**